File: RadStudio_XE6_Demo/radstudiodemos-code-1932-branches-RadStudio_XE6/Object Pascal/DataSnap/connectors_many_devicies/WindowsPhone7Clients/CompanyTweetClient/CompanyTweetClient/proxy/DSProxy.cs
// 
// Created by the DataSnap proxy generator.
// 7/22/2011 2:43:10 PM
// 

using System;
using System.Threading;

namespace Embarcadero.Datasnap.WindowsPhone7
{
  public class DSProxy
  {
    public class TCompanyTweet : DSAdmin
    {
      public TCompanyTweet(DSRESTConnection Connection, ExceptionCallback ExCal)
          : base(Connection, ExCal)
      {
      }

      private DSRESTParameterMetaData[] TCompanyTweet_Logout_Metadata;
      private DSRESTParameterMetaData[] get_TCompanyTweet_Logout_Metadata()
      {
        if (TCompanyTweet_Logout_Metadata == null)
        {
          TCompanyTweet_Logout_Metadata = new DSRESTParameterMetaData[]
          {
          };
        }
        return TCompanyTweet_Logout_Metadata;
      }

      public delegate void LogoutCallback();

      public void Logout(LogoutCallback callback = null, ExceptionCallback ExCal = null)
      {
        DSRESTCommand cmd = getConnection().CreateCommand();
        cmd.setRequestType(DSHTTPRequestType.GET);
        cmd.setText("TCompanyTweet.Logout");
        cmd.prepare(get_TCompanyTweet_Logout_Metadata());
        InternalConnectionDelegate LogoutDel = () =>
        {
          if (callback != null)
          {
            try
            {
              callback.DynamicInvoke();
            }
            catch (Exception ex)
            {
              if (ExCal != null) getConnection().syncContext.Send(new SendOrPostCallback(x => ExCal.DynamicInvoke(ex.InnerException)), null);
              else getConnection().syncContext.Send(new SendOrPostCallback(x => BaseExCal.DynamicInvoke(ex.InnerException)), null);
            }
          }
        };
        getConnection().execute(cmd, this, LogoutDel, ExCal);
      }
      private DSRESTParameterMetaData[] TCompanyTweet_LoginUser_Metadata;
      private DSRESTParameterMetaData[] get_TCompanyTweet_LoginUser_Metadata()
      {
        if (TCompanyTweet_LoginUser_Metadata == null)
        {
          TCompanyTweet_LoginUser_Metadata = new DSRESTParameterMetaData[]
          {
            new DSRESTParameterMetaData("UserName", DSRESTParamDirection.Input, DBXDataTypes.WideStringType, "string"),
            new DSRESTParameterMetaData("ReturnMessage", DSRESTParamDirection.Output, DBXDataTypes.WideStringType, "string"),
            new DSRESTParameterMetaData("", DSRESTParamDirection.ReturnValue, DBXDataTypes.BooleanType, "Boolean"),
          };
        }
        return TCompanyTweet_LoginUser_Metadata;
      }

      /**
       * @param UserName [in] - Type on server: string
       * @param ReturnMessage [out] - Type on server: string
       * @return result - Type on server: Boolean
       */
      public class LoginUserReturns
      {
        public String ReturnMessage;
        public bool returnValue;
      }
      public delegate void LoginUserCallback(LoginUserReturns Returns);

      public void LoginUser(String UserName, LoginUserCallback callback = null, ExceptionCallback ExCal = null)
      {
        DSRESTCommand cmd = getConnection().CreateCommand();
        cmd.setRequestType(DSHTTPRequestType.GET);
        cmd.setText("TCompanyTweet.LoginUser");
        cmd.prepare(get_TCompanyTweet_LoginUser_Metadata());
        InternalConnectionDelegate LoginUserDel = () =>
        {
          if (callback != null)
          {
            try
            {
              LoginUserReturns ret = new LoginUserReturns();
              ret.ReturnMessage = cmd.getParameter(1).getValue().GetAsString();
              ret.returnValue = cmd.getParameter(2).getValue().GetAsBoolean();
              callback.DynamicInvoke(ret);
            }
            catch (Exception ex)
            {
              if (ExCal != null) getConnection().syncContext.Send(new SendOrPostCallback(x => ExCal.DynamicInvoke(ex.InnerException)), null);
              else getConnection().syncContext.Send(new SendOrPostCallback(x => BaseExCal.DynamicInvoke(ex.InnerException)), null);
            }
          }
        };
        cmd.getParameter(0).getValue().SetAsString(UserName);
        getConnection().execute(cmd, this, LoginUserDel, ExCal);
      }
      private DSRESTParameterMetaData[] TCompanyTweet_UsersList_Metadata;
      private DSRESTParameterMetaData[] get_TCompanyTweet_UsersList_Metadata()
      {
        if (TCompanyTweet_UsersList_Metadata == null)
        {
          TCompanyTweet_UsersList_Metadata = new DSRESTParameterMetaData[]
          {
            new DSRESTParameterMetaData("", DSRESTParamDirection.ReturnValue, DBXDataTypes.JsonValueType, "TJSONArray"),
          };
        }
        return TCompanyTweet_UsersList_Metadata;
      }

      /**
       * @return result - Type on server: TJSONArray
       */
      public delegate void UsersListCallback(TJSONArray Result);

      public void UsersList(UsersListCallback callback = null, ExceptionCallback ExCal = null)
      {
        DSRESTCommand cmd = getConnection().CreateCommand();
        cmd.setRequestType(DSHTTPRequestType.GET);
        cmd.setText("TCompanyTweet.UsersList");
        cmd.prepare(get_TCompanyTweet_UsersList_Metadata());
        InternalConnectionDelegate UsersListDel = () =>
        {
          if (callback != null)
          {
            try
            {
              callback.DynamicInvoke((TJSONArray)cmd.getParameter(0).getValue().GetAsJSONValue());
            }
            catch (Exception ex)
            {
              if (ExCal != null) getConnection().syncContext.Send(new SendOrPostCallback(x => ExCal.DynamicInvoke(ex.InnerException)), null);
              else getConnection().syncContext.Send(new SendOrPostCallback(x => BaseExCal.DynamicInvoke(ex.InnerException)), null);
            }
          }
        };
        getConnection().execute(cmd, this, UsersListDel, ExCal);
      }
      private DSRESTParameterMetaData[] TCompanyTweet_ConnectedUsers_Metadata;
      private DSRESTParameterMetaData[] get_TCompanyTweet_ConnectedUsers_Metadata()
      {
        if (TCompanyTweet_ConnectedUsers_Metadata == null)
        {
          TCompanyTweet_ConnectedUsers_Metadata = new DSRESTParameterMetaData[]
          {
            new DSRESTParameterMetaData("", DSRESTParamDirection.ReturnValue, DBXDataTypes.JsonValueType, "TJSONArray"),
          };
        }
        return TCompanyTweet_ConnectedUsers_Metadata;
      }

      /**
       * @return result - Type on server: TJSONArray
       */
      public delegate void ConnectedUsersCallback(TJSONArray Result);

      public void ConnectedUsers(ConnectedUsersCallback callback = null, ExceptionCallback ExCal = null)
      {
        DSRESTCommand cmd = getConnection().CreateCommand();
        cmd.setRequestType(DSHTTPRequestType.GET);
        cmd.setText("TCompanyTweet.ConnectedUsers");
        cmd.prepare(get_TCompanyTweet_ConnectedUsers_Metadata());
        InternalConnectionDelegate ConnectedUsersDel = () =>
        {
          if (callback != null)
          {
            try
            {
              callback.DynamicInvoke((TJSONArray)cmd.getParameter(0).getValue().GetAsJSONValue());
            }
            catch (Exception ex)
            {
              if (ExCal != null) getConnection().syncContext.Send(new SendOrPostCallback(x => ExCal.DynamicInvoke(ex.InnerException)), null);
              else getConnection().syncContext.Send(new SendOrPostCallback(x => BaseExCal.DynamicInvoke(ex.InnerException)), null);
            }
          }
        };
        getConnection().execute(cmd, this, ConnectedUsersDel, ExCal);
      }
      private DSRESTParameterMetaData[] TCompanyTweet_SetUsersToFollow_Metadata;
      private DSRESTParameterMetaData[] get_TCompanyTweet_SetUsersToFollow_Metadata()
      {
        if (TCompanyTweet_SetUsersToFollow_Metadata == null)
        {
          TCompanyTweet_SetUsersToFollow_Metadata = new DSRESTParameterMetaData[]
          {
            new DSRESTParameterMetaData("Users", DSRESTParamDirection.Input, DBXDataTypes.JsonValueType, "TJSONArray"),
          };
        }
        return TCompanyTweet_SetUsersToFollow_Metadata;
      }

      /**
       * @param Users [in] - Type on server: TJSONArray
       */
      public delegate void SetUsersToFollowCallback();

      public void SetUsersToFollow(TJSONArray Users, SetUsersToFollowCallback callback = null, ExceptionCallback ExCal = null)
      {
        DSRESTCommand cmd = getConnection().CreateCommand();
        cmd.setRequestType(DSHTTPRequestType.POST);
        cmd.setText("TCompanyTweet.SetUsersToFollow");
        cmd.prepare(get_TCompanyTweet_SetUsersToFollow_Metadata());
        InternalConnectionDelegate SetUsersToFollowDel = () =>
        {
          if (callback != null)
          {
            try
            {
              callback.DynamicInvoke();
            }
            catch (Exception ex)
            {
              if (ExCal != null) getConnection().syncContext.Send(new SendOrPostCallback(x => ExCal.DynamicInvoke(ex.InnerException)), null);
              else getConnection().syncContext.Send(new SendOrPostCallback(x => BaseExCal.DynamicInvoke(ex.InnerException)), null);
            }
          }
        };
        cmd.getParameter(0).getValue().SetAsJSONValue(Users);
        getConnection().execute(cmd, this, SetUsersToFollowDel, ExCal);
      }
      private DSRESTParameterMetaData[] TCompanyTweet_SendTweet_Metadata;
      private DSRESTParameterMetaData[] get_TCompanyTweet_SendTweet_Metadata()
      {
        if (TCompanyTweet_SendTweet_Metadata == null)
        {
          TCompanyTweet_SendTweet_Metadata = new DSRESTParameterMetaData[]
          {
            new DSRESTParameterMetaData("Tweet", DSRESTParamDirection.Input, DBXDataTypes.WideStringType, "string"),
          };
        }
        return TCompanyTweet_SendTweet_Metadata;
      }

      /**
       * @param Tweet [in] - Type on server: string
       */
      public delegate void SendTweetCallback();

      public void SendTweet(String Tweet, SendTweetCallback callback = null, ExceptionCallback ExCal = null)
      {
        DSRESTCommand cmd = getConnection().CreateCommand();
        cmd.setRequestType(DSHTTPRequestType.GET);
        cmd.setText("TCompanyTweet.SendTweet");
        cmd.prepare(get_TCompanyTweet_SendTweet_Metadata());
        InternalConnectionDelegate SendTweetDel = () =>
        {
          if (callback != null)
          {
            try
            {
              callback.DynamicInvoke();
            }
            catch (Exception ex)
            {
              if (ExCal != null) getConnection().syncContext.Send(new SendOrPostCallback(x => ExCal.DynamicInvoke(ex.InnerException)), null);
              else getConnection().syncContext.Send(new SendOrPostCallback(x => BaseExCal.DynamicInvoke(ex.InnerException)), null);
            }
          }
        };
        cmd.getParameter(0).getValue().SetAsString(Tweet);
        getConnection().execute(cmd, this, SendTweetDel, ExCal);
      }
    }

  }
}

File: RadStudio_XE6_Demo/radstudiodemos-code-1932-branches-RadStudio_XE6/Object Pascal/DataSnap/connectors_many_devicies/iOSClients/free_pascal_ios_company_tweet_ios42/DSProxy.pas
// 
// Created by the DataSnap proxy generator.
// 15/12/2011 18:54:14
// 

unit DSProxy ;

interface
  uses
    Classes,
    DB,
    DSAdmin,
    DSRESTParameterMetaData,
    DBXFPCJson,
    DBXValue,
    DBXFPCCommon,
    DSRESTConnection, DSRestTypes;

  type
    TCompanyTweet = class(TDSAdmin)
    Private
      FTCompanyTweet_Logout:TDSRESTParameterMetaDataArray;
      FTCompanyTweet_LoginUser:TDSRESTParameterMetaDataArray;
      FTCompanyTweet_UsersList:TDSRESTParameterMetaDataArray;
      FTCompanyTweet_ConnectedUsers:TDSRESTParameterMetaDataArray;
      FTCompanyTweet_SetUsersToFollow:TDSRESTParameterMetaDataArray;
      FTCompanyTweet_SendTweet:TDSRESTParameterMetaDataArray;
    protected
      function getTCompanyTweet_Logout:TDSRESTParameterMetaDataArray;
      function getTCompanyTweet_LoginUser:TDSRESTParameterMetaDataArray;
      function getTCompanyTweet_UsersList:TDSRESTParameterMetaDataArray;
      function getTCompanyTweet_ConnectedUsers:TDSRESTParameterMetaDataArray;
      function getTCompanyTweet_SetUsersToFollow:TDSRESTParameterMetaDataArray;
      function getTCompanyTweet_SendTweet:TDSRESTParameterMetaDataArray;
    public
      destructor Destroy;override;

      procedure Logout();

      (*
       * @param UserName [in] - Type on server: string
       * @param ReturnMessage [out] - Type on server: string
       * @return result - Type on server: boolean
       *)
      function LoginUser(UserName: string;out ReturnMessage: string):boolean;

      (*
       * @return result - Type on server: TJSONArray
       *)
      function UsersList():TJSONArray;

      (*
       * @return result - Type on server: TJSONArray
       *)
      function ConnectedUsers():TJSONArray;

      (*
       * @param Users [in] - Type on server: TJSONArray
       *)
      procedure SetUsersToFollow(Users: TJSONArray);

      (*
       * @param Tweet [in] - Type on server: string
       *)
      procedure SendTweet(Tweet: string);
    end;

implementation


{TCompanyTweet}

function TCompanyTweet.getTCompanyTweet_Logout:TDSRESTParameterMetaDataArray;
begin
  if Length(FTCompanyTweet_Logout) = 0 then
  begin
    SetLength(Result, 0);
  end
  else
    result:= FTCompanyTweet_Logout;
end;

function TCompanyTweet.getTCompanyTweet_LoginUser:TDSRESTParameterMetaDataArray;
begin
  if Length(FTCompanyTweet_LoginUser) = 0 then
  begin
    SetLength(Result, 3);
    Result[0] := TDSRESTParameterMetaData.CreateParam( 'UserName',Input,WideStringType,'string');
    Result[1] := TDSRESTParameterMetaData.CreateParam( 'ReturnMessage',Output,WideStringType,'string');
    Result[2] := TDSRESTParameterMetaData.CreateParam( '',ReturnValue,BooleanType,'boolean');
  end
  else
    result:= FTCompanyTweet_LoginUser;
end;

function TCompanyTweet.getTCompanyTweet_UsersList:TDSRESTParameterMetaDataArray;
begin
  if Length(FTCompanyTweet_UsersList) = 0 then
  begin
    SetLength(Result, 1);
    Result[0] := TDSRESTParameterMetaData.CreateParam( '',ReturnValue,JsonValueType,'TJSONArray');
  end
  else
    result:= FTCompanyTweet_UsersList;
end;

function TCompanyTweet.getTCompanyTweet_ConnectedUsers:TDSRESTParameterMetaDataArray;
begin
  if Length(FTCompanyTweet_ConnectedUsers) = 0 then
  begin
    SetLength(Result, 1);
    Result[0] := TDSRESTParameterMetaData.CreateParam( '',ReturnValue,JsonValueType,'TJSONArray');
  end
  else
    result:= FTCompanyTweet_ConnectedUsers;
end;

function TCompanyTweet.getTCompanyTweet_SetUsersToFollow:TDSRESTParameterMetaDataArray;
begin
  if Length(FTCompanyTweet_SetUsersToFollow) = 0 then
  begin
    SetLength(Result, 1);
    Result[0] := TDSRESTParameterMetaData.CreateParam( 'Users',Input,JsonValueType,'TJSONArray');
  end
  else
    result:= FTCompanyTweet_SetUsersToFollow;
end;

function TCompanyTweet.getTCompanyTweet_SendTweet:TDSRESTParameterMetaDataArray;
begin
  if Length(FTCompanyTweet_SendTweet) = 0 then
  begin
    SetLength(Result, 1);
    Result[0] := TDSRESTParameterMetaData.CreateParam( 'Tweet',Input,WideStringType,'string');
  end
  else
    result:= FTCompanyTweet_SendTweet;
end;
destructor TCompanyTweet.Destroy;
begin
   TDSRESTParameterMetaData.releaseArray(FTCompanyTweet_Logout);
   TDSRESTParameterMetaData.releaseArray(FTCompanyTweet_LoginUser);
   TDSRESTParameterMetaData.releaseArray(FTCompanyTweet_UsersList);
   TDSRESTParameterMetaData.releaseArray(FTCompanyTweet_ConnectedUsers);
   TDSRESTParameterMetaData.releaseArray(FTCompanyTweet_SetUsersToFollow);
   TDSRESTParameterMetaData.releaseArray(FTCompanyTweet_SendTweet);
  inherited;
end;

procedure TCompanyTweet.Logout();
var
  cmd: TDSRestCommand;
begin

  cmd := Connection.createCommand;
  try
    cmd.RequestType := GET;
    cmd.text := 'TCompanyTweet.Logout';
    cmd.prepare(getTCompanyTweet_Logout);
    
    Connection.execute(cmd);
  finally
    cmd.free;
  end;
end;

(*
 * @param UserName [in] - Type on server: string
 * @param ReturnMessage [out] - Type on server: string
 * @return result - Type on server: boolean
 *)
function TCompanyTweet.LoginUser( username:string;out  returnmessage:string):boolean;
var
  cmd: TDSRestCommand;
begin

  cmd := Connection.createCommand;
  try
    cmd.RequestType := GET;
    cmd.text := 'TCompanyTweet.LoginUser';
    cmd.prepare(getTCompanyTweet_LoginUser);
    cmd.parameters.items[0].Value.AsString:=username;
    
    Connection.execute(cmd);
    returnmessage :=   cmd.parameters.items[1].Value.AsString;
    
    result:= cmd.parameters.items[2].Value.AsBoolean;
  finally
    cmd.free;
  end;
end;

(*
 * @return result - Type on server: TJSONArray
 *)
function TCompanyTweet.UsersList():TJSONArray;
var
  cmd: TDSRestCommand;
begin

  cmd := Connection.createCommand;
  try
    cmd.RequestType := GET;
    cmd.text := 'TCompanyTweet.UsersList';
    cmd.prepare(getTCompanyTweet_UsersList);
    
    Connection.execute(cmd);
    
    result:= cmd.parameters.items[0].Value.AsJSONValue as TJSONArray;
  finally
    cmd.free;
  end;
end;

(*
 * @return result - Type on server: TJSONArray
 *)
function TCompanyTweet.ConnectedUsers():TJSONArray;
var
  cmd: TDSRestCommand;
begin

  cmd := Connection.createCommand;
  try
    cmd.RequestType := GET;
    cmd.text := 'TCompanyTweet.ConnectedUsers';
    cmd.prepare(getTCompanyTweet_ConnectedUsers);
    
    Connection.execute(cmd);
    
    result:= cmd.parameters.items[0].Value.AsJSONValue as TJSONArray;
  finally
    cmd.free;
  end;
end;

(*
 * @param Users [in] - Type on server: TJSONArray
 *)
procedure TCompanyTweet.SetUsersToFollow( users:TJSONArray);
var
  cmd: TDSRestCommand;
begin

  cmd := Connection.createCommand;
  try
    cmd.RequestType := POST;
    cmd.text := 'TCompanyTweet.SetUsersToFollow';
    cmd.prepare(getTCompanyTweet_SetUsersToFollow);
    cmd.parameters.items[0].Value.AsJSONValue:=users;
    
    Connection.execute(cmd);
  finally
    cmd.free;
  end;
end;

(*
 * @param Tweet [in] - Type on server: string
 *)
procedure TCompanyTweet.SendTweet( tweet:string);
var
  cmd: TDSRestCommand;
begin

  cmd := Connection.createCommand;
  try
    cmd.RequestType := GET;
    cmd.text := 'TCompanyTweet.SendTweet';
    cmd.prepare(getTCompanyTweet_SendTweet);
    cmd.parameters.items[0].Value.AsString:=tweet;
    
    Connection.execute(cmd);
  finally
    cmd.free;
  end;
end;
end.
